--- a/src/gs-monitor.c
+++ b/src/gs-monitor.c
@@ -141,7 +141,7 @@
                 /* Add a 1s delay for VT switching.
                  * This seems to preserved content exposure.
                  */
-                g_timeout_add_seconds (1,
+                g_timeout_add_seconds (4,
                                        (GSourceFunc)gs_monitor_lock_session,
                                        monitor);
         } else {
@@ -247,7 +247,7 @@
                 /* Add a 1s delay for VT switching.
                  * This seems to preserved content exposure.
                  */
-                g_timeout_add_seconds (1,
+                g_timeout_add_seconds (4,
                                        (GSourceFunc)gs_monitor_lock_session,
                                        monitor);
         } else {
